kernelci/runtime/kubernetes.py

```diff
@@ -3,28 +3,28 @@
 # Copyright (C) 2022-2023 Collabora Limited
 # Author: Guillaume Tucker <guillaume.tucker@collabora.com>
 
-import kubernetes
+"""Kubernetes runtime implementation"""
+
 import random
 import re
 import string
-from jinja2 import Environment, FileSystemLoader
-from kernelci.runtime import Runtime
+
+import kubernetes
+from . import Runtime
 
 
 class Kubernetes(Runtime):
-    DEFAULT_TEMPLATE_PATHS = ['config/runtime', '/etc/kernelci/runtime']
+    """Runtime implementation to run jobs in a Kubernetes
+
+    This Runtime implementation generates a Kubernetes YAML job definition and
+    submits it or "applies" it in a cluster.
+    """
+
     RANDOM_CHARACTERS = string.ascii_lowercase + string.digits
 
-    def generate(self, params, device_config, plan_config,
-                 callback_opts=None, templates_paths=None):
-        if templates_paths is None:
-            templates_paths = self.DEFAULT_TEMPLATE_PATHS
-        jinja2_env = Environment(loader=FileSystemLoader(templates_paths))
-        template_path = plan_config.get_template_path(None)
-        template = jinja2_env.get_template(template_path)
-        job_name = '-'.join([
-            'kci', params['node_id'], params['name'][:24]
-        ])
+    def generate(self, params, plan_config):
+        template = self._get_template(plan_config)
+        job_name = '-'.join(['kci', params['node_id'], params['name'][:24]])
         safe_name = re.sub(r'[\:/_+=]', '-', job_name).lower()
         rand_sx = ''.join(random.sample(self.RANDOM_CHARACTERS, 8))
```

Comment:
Again, this change hasn't been introduced here, but maybe it's a good idea to change `RANDOM_CHARACTERS` name to something more descriptive - this characters are not really random. Maybe  simply use `(ascii_lowercase+digits)` here.

Reply:
I think `RANDOM_CHARACTERS` is actually better as the set is an implementation detail.  We could call it `RANDOM_K8S_JOB_NAME_CHARACTERS` but that's longer and not very useful for a class that's not used anywhere else.

Reply:
OK so something like `K8S_JOB_NAME_CHARACTERS` would be more descriptive, even though I think that's still not the best name we can come up with.

Reply:
Alright I've renamed it.